### ISSUE TITLE
Adding roles doesn't seem to be effective until the user logs out

### ISSUE THREAD

**Issue by gaazkam** [aspnet/Docs#8474]
This is one tidbit that caught me off-guard. I can't see it mentioned on this docu page. If I might suggest, should this be briefly mentioned in this docs page and perhaps explained somewhere else at lenght?

The thing I noticed is that if a user is added to a role via userManager.AddToRoleAsync(idenityUser, role) while they are logged in, context.User.IsInRole(role) will still return false until the user logs out and in again.

I tried to modify this sample app to allow admins promote regular users to managers. The promotion was not successfull until they logged out. I also tried to enhance the signalR chat app presented in these docs pages to support accounts and to allow mods to mute users via userManager.AddToRoleAsync(idenityUser, "Muted"). This should be enforced immediatelly, not when the user in question logs out.

Is there any docs pages that covers this issue?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b66f8ffa-699c-8b11-0cab-805f5953805f
* Version Independent ID: 7c151ed5-fd78-84fd-a08a-88912990c4ca
* Content: [Create an ASP.NET Core app with user data protected by authorization](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/secure-data?view=aspnetcore-2.2)
* Content Source: [aspnetcore/security/authorization/secure-data.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authorization/secure-data.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@HaoK  please respond.

**Comment by HaoK** [comment]
Yes roles are stored in the cookie, so any changes made to the users roles wouldn't be persisted into the cookie until the next time the cookie is regenerated (typically this happens automatically every 30 minutes or so due to the security stamp validator)

**Comment by gaazkam** [comment]
Is there any way to force a regeneration of this cookie? Invalidating the previous cookie maybe?

**Comment by Rick-Anderson** [comment]
@HaoK 
>  any changes made to the users roles wouldn't be persisted into the cookie until the next time the cookie is regenerated (typically this happens automatically every 30 minutes or so due to the security stamp validator)

Or the user signs out, signs in again. - right?

**Comment by HaoK** [comment]
Yeah, the app should just do that explicitly on the page that added the role(signinManager.RefreshSignIn)
